source/Halibut/ServiceModel/PendingRequestQueueAsync.cs

```diff
@@ -283,16 +291,24 @@ async Task<bool> WaitForResponseToBeSet(TimeSpan timeout, CancellationToken canc
                 }
                 catch (OperationCanceledException ex)
                 {
-                    // Cancel the Request to force cancellation to the socket if it is currently being transferred 
-                    CancelRequest();
-
-                    if (timeoutCancellationTokenSource.IsCancellationRequested)
-                    {
-                        return false;
-                    }
-
                     using (await transferLock.LockAsync(CancellationToken.None))
```

Comment:
Ensure there is no race condition allowing a just cancelled request to be dequeued